### PR TITLE
fix core folder with php-cs-fixer for psr-2

### PR DESCRIPTION
**Pull request by fabiocarneiro**
This fixes all the psr-2 problems at once. 

This was generated using php-cs-fixer with psr-2 level plus unused_use filter.

Here is a list of which problem was found in each file.

```
   1) Plugin/MetadataLoader.php (unused_use)
   2) Plugin/Dependency.php (method_argument_space, braces)
   3) Plugin/Metric.php (braces, eof_ending)
   4) Plugin/AggregatedMetric.php (eof_ending, unused_use)
   5) Plugin/API.php (braces)
   6) Plugin/Widgets.php (braces)
   7) Plugin/Menu.php (braces)
   8) Plugin/Visualization.php (trailing_spaces, elseif)
   9) Plugin/Tasks.php (eof_ending)
  10) Plugin/Settings.php (braces)
  11) Plugin/Dimension/ActionDimension.php (braces, unused_use)
  12) Plugin/Dimension/VisitDimension.php (braces, unused_use)
  13) Plugin/Dimension/ConversionDimension.php (braces, unused_use)
  14) Plugin/Manager.php (parenthesis, braces, unused_use)
  15) Plugin/ComponentFactory.php (eof_ending)
  16) Plugin/ViewDataTable.php (braces)
  17) Plugin/ProcessedMetric.php (eof_ending)
  18) Plugin/ControllerAdmin.php (braces)
  19) Plugin/Controller.php (elseif, braces)
  20) Plugin/Segment.php (braces, eof_ending)
  21) Plugin/Report.php (function_declaration, elseif, braces, eof_ending)
  22) Translation/Loader/JsonFileLoader.php (unused_use)
  23) Translation/Transifex/API.php (elseif, eof_ending)
  24) UrlHelper.php (elseif)
  25) API/ResponseBuilder.php (unused_use)
  26) API/DataTableManipulator/ReportTotalsCalculator.php (braces, unused_use)
  27) API/CORSHandler.php (trailing_spaces)
  28) API/DataTableGenericFilter.php (visibility, eof_ending, unused_use)
  29) API/Proxy.php (braces)
  30) API/Request.php (braces)
  31) API/DataTablePostProcessor.php (eof_ending)
  32) API/ApiRenderer.php (braces)
  33) API/Inconsistencies.php (eof_ending)
  34) API/DataTableManipulator.php (elseif, visibility, unused_use)
  35) BaseFactory.php (trailing_spaces, eof_ending)
  36) Tracker.php (elseif, braces)
  37) ReportRenderer.php (function_call_space, unused_use)
  38) dispatch.php (eof_ending)
  39) Auth.php (eof_ending)
  40) Session/SaveHandler/DbTable.php (visibility)
  41) Cookie.php (braces)
  42) DataArray.php (braces)
  43) Columns/Updater.php (function_call_space, eof_ending)
  44) Columns/Dimension.php (eof_ending, unused_use)
  45) AssetManager/UIAssetMinifier.php (braces)
  46) AssetManager/UIAssetMerger/JScriptUIAssetMerger.php (visibility, braces)
  47) AssetManager/UIAssetMerger/StylesheetUIAssetMerger.php (parenthesis, visibility)
  48) AssetManager/UIAssetCatalogSorter.php (function_declaration, visibility, braces)
  49) AssetManager/UIAssetFetcher/StaticUIAssetFetcher.php (visibility, braces)
  50) AssetManager/UIAssetFetcher/JScriptUIAssetFetcher.php (braces)
  51) AssetManager/UIAssetMerger.php (visibility, braces, unused_use)
  52) AssetManager/UIAssetCatalog.php (visibility)
  53) AssetManager/UIAssetFetcher.php (visibility, braces)
  54) AssetManager/UIAsset/OnDiskUIAsset.php (visibility, function_call_space, braces)
  55) DataAccess/ArchiveWriter.php (braces)
  56) DataAccess/RawLogDao.php (eof_ending)
  57) DataAccess/TableMetadata.php (eof_ending)
  58) DataAccess/LogQueryBuilder.php (trailing_spaces, elseif, line_after_namespace, braces, eof_ending)
  59) DataAccess/ArchiveTableCreator.php (unused_use)
  60) DataAccess/LogAggregator.php (braces)
  61) DataAccess/Model.php (braces)
  62) DataAccess/ArchiveSelector.php (braces)
  63) DataAccess/Actions.php (eof_ending)
  64) DeviceDetectorCache.php (braces, unused_use)
  65) Application/Kernel/EnvironmentValidator.php (eof_ending, unused_use)
  66) Application/Kernel/PluginList.php (eof_ending)
  67) Application/Kernel/GlobalSettingsProvider.php (eof_ending)
  68) Application/Environment.php (eof_ending)
  69) Mail.php (lowercase_constants)
  70) Singleton.php (braces)
  71) Unzip.php (braces)
  72) DataFiles/SearchEngines.php (indentation, trailing_spaces)
  73) Metrics.php (method_argument_space, braces)
  74) ReportRenderer/Html.php (braces)
  75) ReportRenderer/Pdf.php (elseif, braces)
  76) Profiler.php (function_declaration, parenthesis, method_argument_space, braces, unused_use)
  77) CronArchive.php (parenthesis, braces, eof_ending)
  78) EventDispatcher.php (elseif, braces, eof_ending, unused_use)
  79) Url.php (visibility, single_line_after_imports, braces, unused_use)
  80) Updates.php (visibility)
  81) DataTable.php (elseif, braces)
  82) Plugin.php (braces)
  83) FrontController.php (method_argument_space, unused_use)
  84) CliMulti.php (trailing_spaces, parenthesis, lowercase_constants, braces)
  85) ViewDataTable/Config.php (line_after_namespace)
  86) ViewDataTable/Request.php (braces)
  87) ViewDataTable/RequestConfig.php (line_after_namespace, unused_use)
  88) Session.php (elseif)
  89) Version.php (braces)
  90) Date.php (braces)
  91) Filesystem.php (elseif, braces)
  92) Twig.php (indentation, function_declaration, parenthesis, method_argument_space, braces)
  93) Site.php (braces)
  94) Updater.php (parenthesis, visibility, unused_use)
  95) Segment/SegmentExpression.php (visibility)
  96) CliMulti/Output.php (braces)
  97) CliMulti/RequestCommand.php (unused_use)
  98) CliMulti/CliPhp.php (braces, unused_use)
  99) CliMulti/Process.php (visibility, braces)
 100) Archive.php (elseif, unused_use)
 101) Timer.php (line_after_namespace)
 102) Db/Settings.php (line_after_namespace)
 103) Db/Schema/Mysql.php (function_declaration, braces)
 104) Db/BatchInsert.php (indentation, braces, unused_use)
 105) Db/Adapter.php (braces)
 106) Menu/Group.php (braces)
 107) Menu/MenuReporting.php (line_after_namespace, braces)
 108) Menu/MenuTop.php (line_after_namespace)
 109) Menu/MenuAbstract.php (elseif, braces)
 110) View/RenderTokenParser.php (eof_ending)
 111) View/ViewInterface.php (visibility)
 112) View/UIControl.php (eof_ending)
 113) View.php (trailing_spaces)
 114) Container/ContainerFactory.php (unused_use)
 115) Container/ContainerDoesNotExistException.php (eof_ending)
 116) Period/Factory.php (unused_use)
 117) Period/Year.php (visibility)
 118) Period/Range.php (elseif, braces)
 119) Period/Month.php (braces)
 120) Metrics/Formatter/Html.php (eof_ending)
 121) Metrics/Formatter.php (eof_ending)
 122) Development.php (braces)
 123) CronArchive/FixedSiteIds.php (braces, eof_ending, unused_use)
 124) CronArchive/SharedSiteIds.php (braces, eof_ending)
 125) CronArchive/SitesToReprocessDistributedList.php (eof_ending)
 126) CronArchive/SegmentArchivingRequestUrlProvider.php (elseif, eof_ending)
 127) Access.php (elseif, unused_use)
 128) TCPDF.php (visibility)
 129) DeviceDetectorFactory.php (braces)
 130) CacheId.php (unused_use)
 131) Config/IniFileChain.php (function_declaration, parenthesis, elseif, function_call_space, braces, eof_ending)
 132) Http.php (elseif, braces, eof_ending)
 133) ArchiveProcessor/Loader.php (line_after_namespace, braces, eof_ending, unused_use)
 134) ArchiveProcessor/Rules.php (eof_ending)
 135) ArchiveProcessor/PluginsArchiver.php (parenthesis, braces, unused_use)
 136) Updates/1.7.2-rc7.php (visibility)
 137) Updates/1.10-b4.php (visibility)
 138) Updates/2.11.0-b5.php (visibility)
 139) Updates/2.11.1-b4.php (visibility)
 140) Updates/2.0-b13.php (braces)
 141) Updates/0.2.10.php (visibility)
 142) Updates/1.9.3-b3.php (visibility)
 143) Updates/2.0-a7.php (visibility)
 144) Updates/2.4.0-b3.php (braces)
 145) Updates/2.4.0-b4.php (braces)
 146) Updates/0.2.33.php (visibility)
 147) Updates/2.7.0-b4.php (visibility, braces, eof_ending)
 148) Updates/2.4.0-b1.php (braces)
 149) Updates/2.0-a13.php (braces, unused_use)
 150) Updates/2.0-b3.php (visibility)
 151) Updates/2.0-rc1.php (braces)
 152) Updates/1.9-b19.php (visibility, eof_ending)
 153) Updates/2.0-a17.php (braces, unused_use)
 154) Updates/2.0.4-b7.php (visibility, unused_use)
 155) Updates/1.5-b4.php (visibility)
 156) Updates/1.9.3-b10.php (visibility)
 157) Updates/0.6-rc1.php (visibility, eof_ending)
 158) Updates/1.4-rc1.php (visibility)
 159) Updates/1.9.3-b8.php (visibility)
 160) Updates/1.7-b1.php (visibility)
 161) Updates/1.2.3.php (visibility, eof_ending)
 162) Updates/1.2-rc1.php (visibility, braces, eof_ending)
 163) Updates/0.4.2.php (visibility)
 164) Updates/2.9.0-b1.php (visibility, lowercase_keywords, braces)
 165) Updates/1.2.5-rc1.php (visibility, eof_ending)
 166) Updates/0.2.37.php (visibility)
 167) Updates/1.9-b9.php (visibility, eof_ending)
 168) Updates/2.6.0-b1.php (visibility)
 169) Updates/1.8.3-b1.php (visibility, braces)
 170) Updates/1.9.1-b2.php (visibility)
 171) Updates/1.11-b1.php (visibility)
 172) Updates/1.10.2-b1.php (visibility)
 173) Updates/2.7.0-b2.php (visibility, braces, eof_ending)
 174) Updates/2.13.0-b3.php (braces, unused_use)
 175) Updates/2.0.4-b8.php (visibility, braces, unused_use)
 176) Updates/0.2.13.php (visibility)
 177) Updates/1.2-rc2.php (visibility, eof_ending)
 178) Updates/2.10.0-b4.php (visibility, braces)
 179) Updates/2.9.0-b7.php (visibility, braces)
 180) Updates/1.10.2-b2.php (visibility)
 181) Updates/0.4.4.php (visibility)
 182) Updates/0.5.4.php (visibility)
 183) Updates/1.12-b15.php (visibility)
 184) Updates/1.4-rc2.php (visibility)
 185) Updates/2.0.3-b7.php (visibility, braces)
 186) Updates/2.10.0-b7.php (function_declaration, visibility, braces)
 187) Updates/0.6.3.php (visibility)
 188) Updates/2.11.0-b4.php (function_declaration, visibility, braces)
 189) Updates/0.5.5.php (visibility, braces)
 190) Updates/2.0.4-b5.php (visibility, braces)
 191) Updates/1.2.5-rc7.php (visibility, eof_ending)
 192) Updates/1.10.1.php (visibility)
 193) Updates/2.10.0-b5.php (function_declaration, visibility, lowercase_keywords, braces, unused_use)
 194) Updates/2.5.0-rc2.php (visibility)
 195) Updates/0.2.24.php (visibility)
 196) Updates/0.2.35.php (visibility)
 197) Updates/0.2.32.php (visibility)
 198) Updates/1.6-b1.php (visibility)
 199) Updates/1.5-b2.php (visibility)
 200) Updates/2.1.1-b11.php (visibility)
 201) Updates/2.4.0-b8.php (visibility)
 202) Updates/1.9-b16.php (visibility, eof_ending)
 203) Updates/2.4.0-b6.php (braces)
 204) Updates/2.2.3-b6.php (braces)
 205) Updates/2.2.0-b15.php (visibility, unused_use)
 206) Updates/1.5-b1.php (visibility)
 207) Updates/2.3.0-rc2.php (braces)
 208) Updates/1.5-b5.php (visibility)
 209) Updates/0.5.php (visibility)
 210) Updates/0.9.1.php (visibility)
 211) Updates/2.5.0-b1.php (braces)
 212) Updates/2.10.0-b8.php (visibility, braces)
 213) Updates/2.10.0-b10.php (function_declaration, visibility, braces)
 214) Updates/1.7.2-rc5.php (visibility)
 215) Updates/2.5.0-rc4.php (visibility)
 216) Updates/2.13.1.php (visibility)
 217) Updates/1.12-b16.php (visibility)
 218) Updates/1.5-rc6.php (visibility, eof_ending)
 219) Updates/2.14.0-b1.php (braces, unused_use)
 220) Updates/2.0-b9.php (visibility)
 221) Updates/0.2.27.php (visibility)
 222) Updates/1.5-b3.php (visibility)
 223) Updates/0.7.php (visibility)
 224) Updates/1.12-b1.php (visibility, braces)
 225) Updates/0.4.php (visibility)
 226) Updates/1.6-rc1.php (visibility, eof_ending)
 227) Updates/0.2.12.php (visibility)
 228) Updates/1.1.php (visibility)
 229) Updates/2.11.0-b2.php (visibility, lowercase_keywords, braces)
 230) Updates/2.0-b10.php (visibility)
 231) Updates/0.4.1.php (visibility)
 232) Updates/1.8.4-b1.php (visibility)
 233) SettingsPiwik.php (braces)
 234) DataTable/Renderer.php (braces)
 235) DataTable/Renderer/Xml.php (elseif, eof_ending)
 236) DataTable/Renderer/Php.php (elseif, unused_use)
 237) DataTable/Renderer/Json.php (braces)
 238) DataTable/Renderer/Html.php (braces)
 239) DataTable/Renderer/Rss.php (visibility)
 240) DataTable/Renderer/Csv.php (elseif, braces)
 241) DataTable/Renderer/Console.php (braces, unused_use)
 242) DataTable/Renderer/Tsv.php (visibility)
 243) DataTable/Filter/PatternRecursive.php (braces, unused_use)
 244) DataTable/Filter/ColumnCallbackReplace.php (braces)
 245) DataTable/Filter/Sort.php (visibility, braces)
 246) DataTable/Filter/GroupBy.php (braces)
 247) DataTable/Filter/AddSegmentValue.php (braces)
 248) DataTable/Filter/AddSegmentByLabel.php (elseif, braces)
 249) DataTable/Filter/AddColumnsProcessedMetricsGoal.php (eof_ending, unused_use)
 250) DataTable/Filter/ReplaceSummaryRowLabel.php (unused_use)
 251) DataTable/Filter/CalculateEvolutionFilter.php (braces)
 252) DataTable/Filter/ExcludeLowPopulation.php (unused_use)
 253) DataTable/Filter/ColumnDelete.php (braces)
 254) DataTable/Filter/AddColumnsProcessedMetrics.php (eof_ending, unused_use)
 255) DataTable/Filter/BeautifyTimeRangeLabels.php (elseif)
 256) DataTable/Filter/ColumnCallbackAddColumnQuotient.php (elseif)
 257) DataTable/Filter/ColumnCallbackAddMetadata.php (braces, eof_ending)
 258) DataTable/Filter/PivotByDimension.php (eof_ending)
 259) DataTable/Manager.php (braces)
 260) DataTable/Row.php (elseif, braces, eof_ending)
 261) DataTable/Bridges.php (eof_ending)
 262) DataTable/TableNotFoundException.php (braces)
 263) DataTable/BaseFilter.php (unused_use)
 264) QuickForm2.php (visibility)
 265) Settings/Setting.php (line_after_namespace)
 266) Settings/Manager.php (braces)
 267) Settings/Storage.php (line_after_namespace, braces)
 268) Settings/UserSetting.php (line_after_namespace, braces)
 269) Settings/Storage/Factory.php (braces)
 270) Settings/Storage/StaticStorage.php (line_after_namespace, braces)
 271) Scheduler/Schedule/Weekly.php (function_call_space)
 272) Scheduler/Schedule/Hourly.php (function_call_space)
 273) Scheduler/Schedule/Monthly.php (function_call_space, braces)
 274) Scheduler/Schedule/Daily.php (function_call_space)
 275) Scheduler/Schedule/Schedule.php (function_call_space)
 276) Scheduler/TaskLoader.php (braces)
 277) testMinimumPhpVersion.php (braces)
 278) Piwik.php (elseif, braces, unused_use)
 279) AssetManager.php (visibility, braces, unused_use)
 280) Exception/AuthenticationFailedException.php (eof_ending)
 281) Exception/InvalidRequestParameterException.php (eof_ending)
 282) Exception/MissingFilePermissionException.php (eof_ending)
 283) Exception/UnexpectedWebsiteFoundException.php (eof_ending)
 284) Exception/NoPrivilegesException.php (eof_ending)
 285) Exception/Exception.php (eof_ending)
 286) Exception/NoWebsiteFoundException.php (eof_ending)
 287) Exception/DatabaseSchemaIsNewerThanCodebaseException.php (braces, eof_ending)
 288) Updater/UpdateObserver.php (eof_ending)
 289) DbHelper.php (eof_ending, unused_use)
 290) Filechecks.php (elseif, braces)
 291) Concurrency/DistributedList.php (eof_ending)
 292) ProxyHttp.php (elseif)
 293) Archive/DataTableFactory.php (eof_ending)
 294) Archive/DataCollection.php (elseif, braces)
 295) Archive/ArchivePurger.php (eof_ending, unused_use)
 296) Archive/Parameters.php (eof_ending)
 297) Archive/ArchiveInvalidator.php (trailing_spaces, function_declaration, braces, eof_ending, unused_use)
 298) ArchiveProcessor.php (parenthesis, braces, eof_ending, unused_use)
 299) Segment.php (braces, eof_ending)
 300) Db.php (unused_use)
 301) Theme.php (parenthesis, method_argument_space, braces)
 302) Common.php (function_declaration, elseif, braces, unused_use)
 303) Tracker/Action.php (braces)
 304) Tracker/GoalManager.php (braces)
 305) Tracker/Response.php (function_declaration, braces)
 306) Tracker/Visitor.php (parenthesis)
 307) Tracker/PageUrl.php (indentation, braces, eof_ending)
 308) Tracker/Settings.php (visibility, braces, eof_ending)
 309) Tracker/ScheduledTasksRunner.php (line_after_namespace, unused_use)
 310) Tracker/Handler.php (braces)
 311) Tracker/Visit.php (parenthesis, braces)
 312) Tracker/Db/Pdo/Mysql.php (parenthesis, elseif)
 313) Tracker/Db/Mysqli.php (parenthesis, elseif, braces)
 314) Tracker/VisitExcluded.php (unused_use)
 315) Tracker/Request.php (parenthesis, braces)
 316) Tracker/Visit/Factory.php (line_after_namespace)
 317) Tracker/Visit/ReferrerSpamFilter.php (braces)
 318) Tracker/VisitorNotFoundInDb.php (eof_ending)
 319) Tracker/Handler/Factory.php (braces)
 320) Tracker/RequestSet.php (braces)
 321) Tracker/ActionPageview.php (visibility, braces, unused_use)
 322) Tracker/SettingsStorage.php (braces)
 323) Tracker/Cache.php (visibility, unused_use)
 324) Tracker/Model.php (braces)
 325) Tracker/TableLogAction.php (braces, eof_ending, unused_use)
 326) Console.php (elseif, method_argument_space, braces)
 327) Http/ControllerResolver.php (unused_use)
```

### CODE REVIEW

**Comment by tsteur** [comment]
Is there a way on Github to see the full diff? Does anyone know that?

**Comment by diosmosis** [comment]
https://patch-diff.githubusercontent.com/raw/piwik/piwik/pull/7956.diff I think will work.

**Comment by tsteur** [comment]
Thx @diosmosis .

It looks good to me in general. Especially was interested in making sure nothing changed re `==` and `===` to prevent possible security regressions etc.

It does sometimes remove quite a few lines that I would maybe personally rather prefer to keep. Eg 

```diff
         if ($pluginName) {
-
             if ($this->pluginContainsJScriptAssets($pluginName)) {
-
                 if (Manager::getInstance()->isPluginBundledWithCore($pluginName)) {
-
                     $assetsToRemove[] = $this->getMergedCoreJSAsset();
-
                 } else {
-
                     $assetsToRemove[] = $this->getMergedNonCoreJSAsset();
                 }
             }
-
         } else {
-
             $assetsToRemove[] = $this->getMergedCoreJSAsset();
             $assetsToRemove[] = $this->getMergedNonCoreJSAsset();
         }
```

Sometimes such lines are really good to structure code and make it better readable. Is it maybe possible to ignore new lines? Anyone else has any thoughts on that?

**Comment by fabiocarneiro** [comment]
It might be configurable. But I wouldn't recommend it. It would still be valid PSR-2 code with that lines, but PSR recommends adding lines ONLY to separate the code in blocks, which is not happening in this example.

Much better approach is to use early returns and completely remove all else statements.

```php
if (! $pluginName) {
    $assetsToRemove[] = $this->getMergedCoreJSAsset();
    return;
}

if ($this->pluginContainsJScriptAssets($pluginName)) {
    return;
}

if (Manager::getInstance()->isPluginBundledWithCore($pluginName)) {
    $assetsToRemove[] = $this->getMergedCoreJSAsset();
}

$assetsToRemove[] = $this->getMergedNonCoreJSAsset();
```

**Comment by tsteur** [comment]
The code looks nicer that way! It was only an example though. If we remove it there, we also remove it where it makes sense and where it is good to have it. Adding it afterwards won't happen. I'd say we should rather leave the new lines there and remove the unneeded ones on the fly when working on a file anyway.

Any other opinions on that?

**Comment by fabiocarneiro** [comment]
By removing that empty lines you will be removing them where they shouldn't be. Also probably there are many cases when the removal won't "affect readability".

If you simple ignore that, you may never remember to do that. I would prefer to let the tool fix them now, since it's the default setting for psr-2 and work on the code when you guys find a readability problem. Also, the code will still be readable even without those lines.

**Comment by fabiocarneiro** [comment]
I rebased it again.  Please someone merge this. Since, there are many files it will get conflicts very fast.

File: core/API/DataTableManipulator/ReportTotalsCalculator.php [hunk 1]
```diff
@@ -89,7 +88,8 @@ protected function manipulateDataTable($dataTable)
 
         $metricNames = array();
         foreach ($metricsToCalculate as $metricId) {
-            $metricNames[$metricId] = Metrics::getReadableColumnName($metricId);;
+            $metricNames[$metricId] = Metrics::getReadableColumnName($metricId);
+            ;
```
Comment (mattab) [inline core/API/DataTableManipulator/ReportTotalsCalculator.php]: this extra `;` could be removed

File: core/Menu/Group.php [hunk 2]
```diff
@@ -21,7 +21,8 @@ public function add($subTitleMenu, $url, $tooltip = false)
             'name'    => $subTitleMenu,
             'url'     => $url,
             'tooltip' => $tooltip
-        );;
+        );
+        ;
```
Comment (mattab) [inline core/Menu/Group.php]: extra semi colon

**Comment by mattab** [comment]
Note: the build is failing for this PR, maybe the rebase went wrong? https://travis-ci.org/piwik/piwik/jobs/64199263

I'm not sure what happened there but FYI this build failure is related to the code changed in #7992

Once build is green It's +1 for me to merge

**Comment by fabiocarneiro** [comment]
@mattab I don't know why the build is failing since these are only code style fixes. To make sure I reverted the commit, and ran the command again, but this build errors seems to be random.

I guess this is ok to merge. But it would be nice to investigate why this "Piwik\Tests\System\DuplicateActionsTest". I've already seen it in many builds.

**Comment by mattab** [comment]
Thanks @fabiocarneiro for the PR :+1:

**Comment by mattab** [comment]
See also our follow up issue `Automatically convert our code base to use a consistent code style` #9545

**Pull request merged**